processor/src/main/java/org/mapstruct/ap/internal/model/ObjectFactoryMethodResolver.java

```diff
@@ -121,4 +135,27 @@ private static MethodReference findBuilderFactoryMethod(Type targetType) {
         );
     }
 
+    private static List<SourceMethod> getAllAvailableMethods(Method method, List<SourceMethod> sourceModelMethods) {
+        ParameterProvidedMethods contextProvidedMethods = method.getContextProvidedMethods();
+        if ( contextProvidedMethods.isEmpty() ) {
+            return sourceModelMethods;
+        }
+
+        List<SourceMethod> methodsProvidedByParams = contextProvidedMethods
+            .getAllProvidedMethodsInParameterOrder( method.getContextParameters() );
+
+        List<SourceMethod> availableMethods =
+            new ArrayList<SourceMethod>( methodsProvidedByParams.size() + sourceModelMethods.size() );
+
+        // add only methods from context that do have the @ObjectFactory annotation
+        for ( SourceMethod methodProvidedByParams : methodsProvidedByParams ) {
+            if ( methodProvidedByParams.isObjectFactory() && methodProvidedByParams.hasObjectFactoryAnnotation() ) {
```

Comment:
this looks like a task of the `MethodFamiliySelector`. Need to analyse whether to do it here or leave it up to this selector.